src/change_stream/mod.rs

```diff
@@ -0,0 +1,134 @@
+//! Contains the functionality for ChangeStreams.
+pub mod document;
+pub mod option;
+
+use std::{marker::PhantomData, pin::Pin, task::Poll};
+
+use bson::Document;
+use futures::{stream::Stream, task::Context};
+use serde::{de::DeserializeOwned, Deserialize};
+
+use crate::{
+    change_stream::{
+        document::{ChangeStreamEventDocument, ResumeToken},
+        option::ChangeStreamOptions,
+    },
+    error::Result,
+    options::AggregateOptions,
+    selection_criteria::ReadPreference,
+    Client,
+    Collection,
+    Cursor,
+    Database,
+};
+
+/// A `ChangeStream` streams the ongoing changes of its associated collection, database or
+/// deployment. `ChangeStream` instances should be created with method `watch` against the relevant
+/// target.
+///
+/// `ChangeStream`'s are "resumable", meaning that they can be restarted at a given place in the
+/// stream of events. This is done automatically when the `ChangeStream` encounters certain
+/// ["resumable"](https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.rst#resumable-error) errors, such as transient network failures. It can also be done manually by passing
+/// a [`ResumeToken`](document/struct.ResumeToken.html) retrieved from a past event
+/// into either the
+/// [`resume_after`](option/struct.ChangeStreamOptions.html#structfield.resume_after)
+/// or [`start_after`](option/struct.ChangeStreamOptions.html#structfield.start_after)
+/// (4.2+) options used to create the `ChangeStream`. Issuing a raw change stream aggregation is
+/// discouraged unless users wish to explicitly opt out of resumability.
+///
+/// A `ChangeStream` can be iterated to return batches of instances of any type that implements the
+/// Deserialize trait. By default, it returns batches of
+/// [`ChangeStreamEventDocument`](document/struct.ChangeStreamEventDocument.html).
+/// These documents correspond to changes in the associated collection, database or deployment.
+///
+/// A `ChangeStream` can be iterated like any other `Iterator`:
+///
+/// ```rust
+/// # #[cfg(not(feature = "sync"))]
+/// # use futures::stream::StreamExt;
+/// # use mongodb::{Client, error::Result, bson::doc,
+/// # change_stream::document::ChangeStreamEventDocument};
+/// # #[cfg(feature = "async-std-runtime")]
+/// # use async_std::task;
+/// # #[cfg(feature = "tokio-runtime")]
+/// # use tokio::task;
+/// #
+/// # async fn func() -> Result<()> {
+/// # let client = Client::with_uri_str("mongodb://example.com").await?;
+/// # let coll = client.database("foo").collection("bar");
+/// let mut change_stream = coll.watch(None, None).await?;
+/// let coll_ref = coll.clone();
+/// task::spawn(async move {
+///     coll_ref.insert_one(doc! { "x": 1 }, None).await;
+/// });
+/// while let Some(document) = change_stream.next().await {
+///     println!("operation performed: {:?}, document: {:?}", document.operation_type, document.full_document);
```

Comment:
now that `next` returns a `Document`, this example will need to be updated to manually deserialize the `ChangeStreamEventDocument`.

FYI, you can run the doctests locally without relying on evergreen by doing `cargo test --doc`.